addons/mass_mailing_sale/models/sale.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _name = 'sale.order'
+    _inherit = 'sale.order'
+
+    def _mailing_get_default_domain(self, mailing):
+        """ Exclude by default canceled orders when performing a mass mailing. """
+        return [('state', '!=', 'cancel')]
```

Comment:
Could be nice to test this and the "_sms_get_partner_fields" thingy but I'm guessing it's not easy since the test_mail_full module does not depend on sale.

Reply:
Actually there is already a test for mailing lists domain. I guess we could test each specific implementation of default domain but that would be part of a task "add tests in each sub-mass mailing modules", which would makes sense (testing mailing on all main models) but is a bit out of scope.